parquet/src/main/java/org/apache/iceberg/parquet/ParquetDictionaryRowGroupFilter.java

```diff
@@ -368,6 +368,10 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
       }
 
       Set<T> dictionary = dict(id, lit.comparator());
+      if (dictionary.contains(null)) {
```

Comment:
In the other locations a comment is also included 
      // Allow query engine to make its own decisions regarding SQL 3-valued boolean logic.

Reply:
@kbendick, I don't think that this is a valid case. Parquet never encodes null values in its dictionaries. We can remove this. That avoids the need for the comment, too.

Reply:
I'll remove it 👍